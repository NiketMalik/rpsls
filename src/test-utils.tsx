import { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { THEME } from "constants/theme";

import { StoreProvider } from "context/storeContext";
import { ThemeProvider } from "@emotion/react";

const WithGlobalProviders: FC = ({ children }) => {
  return (
    <StoreProvider>
      <ThemeProvider theme={THEME}>{children}</ThemeProvider>
    </StoreProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithGlobalProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
