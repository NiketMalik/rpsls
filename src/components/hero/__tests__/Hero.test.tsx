import React from "react";
import { render, screen } from "test-utils";

import { Hero } from "../index";

describe("Hero", () => {
  it("renders without error", () => {
    const { container } = render(<Hero />);
    const heroEle = screen.getByTestId("hero");
    expect(heroEle).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
