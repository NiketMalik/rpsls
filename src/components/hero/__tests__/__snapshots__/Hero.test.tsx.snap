// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders without error 1`] = `
<div>
  <div
    class="container"
    data-testid="hero"
  >
    <div
      class="flex-column row"
    >
      <div
        class="col"
      >
        <h1
          class="text-center display-3 font-cursive"
        >
          Rock, Paper, Scissors, Lizard, Spock
        </h1>
      </div>
      <div
        class="col"
      >
        <h2
          class="text-center h4 font-cursive"
        >
          Online dispute resolution the traditional way, from the relative safety of your computer.
        </h2>
      </div>
    </div>
  </div>
</div>
`;
